src/LensPlatformClient.ts

```diff
@@ -145,6 +180,79 @@ class LensPlatformClient {
     };
   }
 
+  /**
+   * Get throwExpected function
+   */
+  get throwExpected() {
+    /**
+     * Executes a given function, catching all exceptions. When an exception is caught
+     * it is converted to strongly-typed `LensPlatformExtension` and thrown again.
+     * @param fn - a function
+     * @param exceptionsMap - map of HTTP error codes onto expected exception creators
+     * @param retryInterval - time between retries
+     * @param retries - How many times to retry
+     * @returns the result of `fn`
+     * @throws expected exceptions or `LensPlatformException` with code 400 if it caught something unexpected
+     * @example
+     * ```
+     * const json = throwExpected(
+     *  () => fetch.get(url),
+     *    {
+     *      404: e => e.url.includes("/user") ?
+     *        new NotFoundException(`User ${username} not found`) :
+     *        new NotFoundException(`Something else not found`),
+     *      500: () => new TokenNotFoundException()
+     *    }
+     * );
+     * ```
+     */
+    const throwExpected = async <T = any>(
+      fn: () => Promise<T>,
+      exceptionsMap: HTTPErrCodeExceptionMap = {},
+    ) => {
+      try {
+        // First try
+        const result = await fn();
+
+        return result;
+      } catch (error: unknown) {
+        // Retry if the error is a Network error and GET
+        if (
+          axios.isAxiosError(error)
+          // If it's an axios error but there is no response => a network error
+          // See https://github.com/axios/axios#handling-errors
+          && !error.response
+          // ONLY retry for GET requests
+          && error.config?.method === "get"
+        ) {
+          if (this.autoRetry) {
+            await timeout(this.retryIntervalMs);
+            try {
+              const retriedResult = await retry<T>(
+                async () =>
+                // Second try
+                  fn()
+                , {
+                  minTimeout: this.retryIntervalMs,
+                  retries: this.retryMaxAttempts - 1 - 1, // Two `-1` because we already did two tries
```

Comment:
If retryMaxAttempts is 2, we should do total of 3 fetches, right? One fetch and 2 retries. Not sure "- 1 - 1" is correct then, looks like "- 1" is enough?